File: app/src/main/java/com/example/josegarcia/todaymeal/adapter/IngredientDiff.java
package com.example.josegarcia.todaymeal.adapter;

import android.support.annotation.NonNull;
import android.support.v7.util.DiffUtil;

import com.example.josegarcia.todaymeal.model.Ingredient;

public class IngredientDiff extends DiffUtil.ItemCallback<Ingredient> {
    @Override
    public boolean areItemsTheSame(@NonNull Ingredient ingredient, @NonNull Ingredient t1) {
        return ingredient.getName().equals(t1.getName());
    }

    @Override
    public boolean areContentsTheSame(@NonNull Ingredient ingredient, @NonNull Ingredient t1) {
        return ingredient.equals(t1);
    }
}

File: app/src/main/java/com/example/josegarcia/todaymeal/adapter/IngredientDiff.kt
package com.example.josegarcia.todaymeal.adapter

import androidx.recyclerview.widget.DiffUtil

import com.example.josegarcia.todaymeal.model.Ingredient

class IngredientDiff : DiffUtil.ItemCallback<Ingredient>() {
    override fun areItemsTheSame(ingredient: Ingredient, t1: Ingredient) =
        ingredient.name == t1.name

    override fun areContentsTheSame(ingredient: Ingredient, t1: Ingredient) =
        ingredient == t1
}
